RFCs/0000-expand-new-task.md

```diff
@@ -0,0 +1,158 @@
+
+- The new `--language` option will require an argument which is the name of the language-specific task template to be used.
+
+- Acceptable values for the `--language` option will initially be `bash`, `powershell`, `ruby`, and `python`.
+
+- The format of future language option values should be a concise, unique, all lowercase language identifier.
+
+- If an unrecognized language option value is specified, PDK will exit with an error.
+
+- If the `--language` option is specified without an argument, PDK will exit with an error.
+
+#### Examples:
+
+```bash
+$ pdk new task my_default_task
+pdk (INFO): Using default task language: 'bash'
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/my_default_task.sh' from template.
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/my_default_task.json' from template.
+```
+
+```bash
+$ pdk new task --language=ruby my_ruby_task
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/my_ruby_task.rb' from template.
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/my_ruby_task.json' from template.
+```
+
+```bash
+$ pdk new task --language=powershell my_ps_task
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/my_ps_task.ps1' from template.
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/my_ps_task.json' from template.
+```
+
+### Add `--from` option to `pdk new task`
+
+- The new `--from` option will require an argument which is an absolute or relative path to an existing script file.
+
+- If the `--from` option is specified without an argument, PDK will exit with an error.
+
+- If the given file does not exist or is not readable, PDK will exit with an error.
+
+- If the given file exists and is readable, PDK will copy it into the `tasks` subdirectory of the current module and write the copy with a name matching the task name supplied to `pdk new task`.
+
+  - For example: `pdk new task --from=/tmp/myscript.sh restart` will copy `/tmp/myscript.sh` to `tasks/restart.sh`.
+
+- For languages that support it (currently only PowerShell?), PDK will attempt to automatically determine the parameters (and output format?) of the source script and use that data to automatically populate the Bolt Task metadata file.
+
+### Add `--private` and `--remote` flags to `pdk new task`
+
+- When specified, these flags will simply pre-populate the generated Bolt Task metadata with `"private": true` and `"remote": true` key/value pairs respectively.
+
+### Add optional `<implementations>` argument to `pdk new task`
+
+- Allow `pdk new task` to accept an unbounded, space-separated list of `implementations` to be provided after the task name.
+
+- When one or more implementation files are provided, PDK will only generate a `tasks/<taskname>.json` file for the new task. The generated metadata will contain an `"implementations"` key with each given implementation file listed.
```

Comment:
`implementations` without `requirements` are misleading as only the first one will be used.

Reply:
There is an item in the "Unresolved Questions" section about whether or not we can or should infer `requirements` from the implementation file name or similar. We could also just template in a `requirements` key with a "NEED_TO_SET" value or something.